Lib/unittest/case.py

```diff
@@ -603,9 +603,21 @@ def _callSetUp(self):
         self.setUp()
 
     def _callTestMethod(self, method):
-        if method() is not None:
-            warnings.warn(f'It is deprecated to return a value that is not None from a '
-                          f'test case ({method})', DeprecationWarning, stacklevel=3)
+        if (result := method()) is not None:
+            import inspect
+            from unittest.async_case import IsolatedAsyncioTestCase
+            msg = (
+                'It is deprecated to return a value that is not None '
+                f'from a test case ({method} returned {type(result).__name__!r})'
+            )
+            if (
+                inspect.iscoroutine(result)
+                and not isinstance(self, IsolatedAsyncioTestCase)
```

Comment:
Is it really needed? Is not `_callTestMethod()` overridden in `IsolatedAsyncioTestCase`?